mantle/cmd/kola/options.go

```diff
@@ -245,10 +245,14 @@ func syncOptionsImpl(useCosa bool) error {
 	if kolaPlatform == "gcp" && kola.GCPOptions.MachineType == "" {
 		switch kola.Options.CosaBuildArch {
 		case "x86_64":
-			if kola.GCPOptions.Confidential {
-				// https://cloud.google.com/compute/confidential-vm/docs/locations
-				fmt.Print("Setting instance type for confidential computing")
-				kola.GCPOptions.MachineType = "n2d-standard-2"
+			if kola.GCPOptions.ConfidentialType != "" {
+				if kola.GCPOptions.ConfidentialType == "SEV" || kola.GCPOptions.ConfidentialType == "SEV_SNP" {
+					// https://cloud.google.com/compute/confidential-vm/docs/locations
+					fmt.Print("Setting instance type for confidential computing")
+					kola.GCPOptions.MachineType = "n2d-standard-2"
+				} else {
+					return fmt.Errorf("Confidential type only supports [SEV, SEV_SNP]")
```

Comment:
I think this error-checking needs to be lowered into `compute.go` instead in the GCP code. So here we can just set the machine type of `ConfidentialType` is non-empty, and leave it to the code later to actually sanity-check it's a valid option.